atc/exec/step_metadata.go

```diff
@@ -61,6 +61,24 @@ func (metadata StepMetadata) Env() []string {
 
 	if metadata.ExternalURL != "" {
 		env = append(env, "ATC_EXTERNAL_URL="+metadata.ExternalURL)
+
+		if metadata.TeamName != "" && metadata.BuildID != 0 {
+			buildURLShort := fmt.Sprintf("%s/builds/%d", metadata.ExternalURL, metadata.BuildID)
+			buildURL := buildURLShort
+
+			if metadata.PipelineName != "" && metadata.JobName != "" && metadata.BuildName != "" {
+				buildURL = fmt.Sprintf("%s/teams/%s/pipelines/%s/jobs/%s/builds/%s",
```

Comment:
This URL would be invalid for instance pipelines. Instance pipelines follow the same URL format, except at the end they have all the instance vars listed as URL params.

For example, the build for one of the checks for this PR is:
```
https://ci.concourse-ci.org/teams/contributor/pipelines/pr/jobs/validate-labels/builds/2?vars.number=9296
```

Note the `?vars.number=9296` at the end. If you strip the instance vars you get a HTTP 404.